client/indexer-db/src/models/file.rs

```diff
@@ -0,0 +1,127 @@
+use chrono::NaiveDateTime;
+use diesel::prelude::*;
+use diesel_async::RunQueryDsl;
+
+use crate::{
+    models::PeerId,
+    schema::{file, file_peer_id},
+    DbConnection,
+};
+
+pub enum FileStorageRequestStep {
+    Requested = 0,
+    Fullfilled = 1,
+    Expired = 2,
+    Revoked = 3,
+}
+
+/// Table that holds the Files (both ongoing requests and completed).
+#[derive(Debug, Queryable, Insertable, Selectable)]
+#[diesel(table_name = file)]
+pub struct File {
```

Comment:
Don't we want to store the BSPs storing this file? It's one of the main reasons for the indexer.

Reply:
Yes, this is stored in the `BspFile` table.